src/main/java/com/good/fly/trip_planner/service/impl/TripServiceImpl.java

```diff
@@ -0,0 +1,163 @@
+        Trip trip = new Trip();
+        Optional<User> optionalUser = userRepository.findById(userId);
+        optionalUser.ifPresent(trip::setUser);
+        trip.setName("Wonder " + trip.getUser().getFirstName() + "'s trip!");
+        Trip trip1 = tripRepository.save(trip);
+        return ResponseEntity
+                .status(HttpStatus.CREATED)
+                .body(trip1);
+    }
+
+    @Override
+    public ResponseEntity<Boolean> deleteTrip(Long tripId) {
+        tripRepository.deleteById(tripId);
+        return ResponseEntity
+                .status(HttpStatus.OK)
+                .body(true);
+    }
+
+    @Override
+    public ResponseEntity<String> setShareStatus(ShareTrip shareTrip) {
+        Optional<Trip> optionalTrip = tripRepository.findById(shareTrip.getTripId());
+        Trip trip = new Trip();
+        if (optionalTrip.isPresent()) {
+            trip = optionalTrip.get();
+            trip.setShared(shareTrip.isShare());
+            tripRepository.save(trip);
+        }
+        return ResponseEntity
+                .status(HttpStatus.OK)
+                .body("Trip " + trip.getName() + " " + (shareTrip.isShare() ? "is shared!" : "is hidden!"));
+    }
+
+    @Override
+    public ResponseEntity<String> setDepartureDate(TripIdDepartureDate tripIdDepartureDate) {
+        Optional<Trip> optionalTrip = tripRepository.findById(tripIdDepartureDate.getTripId());
+        Trip trip = new Trip();
+        if (optionalTrip.isPresent()) {
+            trip = optionalTrip.get();
+            LocalDate departureDate = tripIdDepartureDate.getDepartureDate();
+            int size;
+            if (trip.getPlaces() != null) {
+                size = trip.getPlaces().size();
+            } else size = 0;
+            LocalDate endDate = departureDate.plusDays(size == 0 ? 0 : size * 5);
+            trip.setDepartureDate(departureDate);
+            trip.setEndDate(endDate);
+            tripRepository.save(trip);
+        }
+        return ResponseEntity
+                .status(HttpStatus.OK)
+                .body("Departure and end dates to the trip " + trip.getName() + " is set!");
+
+    }
+
+    @Override
+    public ResponseEntity<List<Trip>> getAllTrips(Long userId) {
+        Optional<User> optionalUser = userRepository.findById(userId);
+        List<Trip> trips = new ArrayList<>();
+        if (optionalUser.isPresent()) {
+            trips = optionalUser.get().getTrips();
+        }
+        return ResponseEntity
+                .status(HttpStatus.OK)
+                .body(trips);
+    }
+
+    @Override
+    public ResponseEntity<String> addPlaceToTrip(Long originalPlaceId, Long tripId) {
+
+        Trip trip = getTrip(tripId);
+
+        OriginalPlace originalPlace = new OriginalPlace();
+        Optional<OriginalPlace> optionalOriginalPlace = originalPlaceRepository.findById(originalPlaceId);
+        if (optionalOriginalPlace.isPresent()) originalPlace = optionalOriginalPlace.get();
+
```

Comment:
what if there are optional isPresentFalse.
below you are trying to get originalPlace.getName()
It's potential NPE

please use orElseThrow() or ifPresent() or ifPresentOrElse()